gemrb/core/Scriptable/Actor.cpp

```diff
@@ -5168,9 +5168,12 @@ void Actor::Die(Scriptable *killer, bool grantXP)
 		// TODO: is this right?
 		killer = area->GetActorByGlobalID(LastHitter);
 	}
-	killer->LastKilled = GetGlobalID();
-	Actor* act = Scriptable::As<Actor>(killer);
```

Comment:
This works if killer is null, so no need to split it up and make it conditional. Just the previous line is problematic.

Reply:
So **area->GetActorByGlobalID(LastHitter);** should return always a correct killer if not it will always fail. Is that correct ?

Reply:
I'm not sure what you're saying. What I'm saying is that `Actor* act = Scriptable::As<Actor>(killer);` is fine regardless of the value of killer.

Reply:
Basically the fix should only change one line.

Reply:
But Actor don't accept **nullptr**
```
matching function for call to ‘GemRB::Actor::As<GemRB::Actor>(std::nullptr_t)’
[build]  5171 |  Actor* act = Scriptable::As<Actor>(nullptr);
```
Will I not be moving the error ?